shared/lsif/ranges.ts

```diff
@@ -193,7 +193,21 @@ export function findOverlappingCodeIntelligenceRange(
                 // right side check
                 (position.line < endLine || (position.line === endLine && position.character < endCharacter))
         ) || null
-    )
+
+    if (overlapping.length === 0) {
+        return null
+    }
+
+    overlapping.sort((a, b) => {
+        const cmp = b.range.start.line - a.range.start.line
+        if (cmp === 0) {
+            return b.range.start.character - a.range.start.character
+        }
+
+        return cmp
+    })
```

Comment:
You may be able to make use of our extension APIs - if you convert these to `sourcegraph.Range`/`sourcegraph.Position` you have methods like `compareTo()`, `isBefore()`, `contains()` etc